core/src/test/scala/unit/kafka/metrics/MetricsTest.scala

```diff
@@ -120,8 +120,15 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
   @ValueSource(strings = Array("zk", "kraft"))
   def testJMXFilter(quorum: String): Unit = {
     // Check if cluster id metrics is not exposed in JMX
-    assertTrue(ManagementFactory.getPlatformMBeanServer
-                 .isRegistered(new ObjectName("kafka.controller:type=KafkaController,name=ActiveControllerCount")))
+    if (quorum == "zk") {
+      assertTrue(ManagementFactory.getPlatformMBeanServer
+        .isRegistered(new ObjectName("kafka.controller:type=KafkaController,name=ActiveControllerCount,brokerId=0")))
+      assertTrue(ManagementFactory.getPlatformMBeanServer
+        .isRegistered(new ObjectName("kafka.controller:type=KafkaController,name=ActiveControllerCount,brokerId=1")))
+    } else {
+      assertTrue(ManagementFactory.getPlatformMBeanServer
+        .isRegistered(new ObjectName("kafka.controller:type=KafkaController,name=ActiveControllerCount")))
+    }
```

Comment:
Noting that all the changes in this file are unrelated to the bug, just collateral from introducing controller metric tags via the KafkaServer in TestUtils.